### ISSUE TITLE
Porting Documentation back to 2.1

### ISSUE THREAD

**Issue by Gfurst** [godotengine/godot-docs#1217]
I a lot of cases I see that big parts of the documentation in 2.1 is missing, while improved a lot in 3, where as a lot of the time they still have the same applicability and functions. Talking about the in-editor documentation here, classes and such.

So shouldn't we consider porting some of this improved documentation back to godot2, for longevity purposes? I know that a lot of devs (myself included) will remain with 2 for a while due to core feature changes in the engine.

I might even be able to do some myself, if the coordinators agree with this. So whats your take on it?

**Comment by akien-mga** [comment]
There's still a lot of work to do on the *master* / 3.x documentation, and 2.1 is going to be retired eventually (sooner than later - 2-3 months after 3.1 is released, so maybe 6 months from now or so), so I wouldn't want documentation contributors to spend resources on that.

Now, if some want to do this work nevertheless, I won't reject a PR adding valid documentation to the 2.1 branch :)

**Comment by Gfurst** [comment]
Its sad that 2.1 its going to get that much better support.
But regardless, in the way I see it, if something is going to be changed (documentation/bugfix/improvement) if it does apply to 3, it does apply to 2.1 as well, with exceptions of course in those specific features/design changes that happened between the two.

So if something does indeed apply to the two of them, they could easily be merged along side to the two of them, if no conflicts arise. Or even merged into one and cherry picked to the other. But of course this will only prove more difficult as differences in code grows with time.

I've also checked the documentation on the engine repo, and it seems to have changed a bit in structure. From having a single xml to multiple files pertaining to something specific. Would this change in structure be easy to backport? It would make backporting more of the engine's documentation, much easier.

**Comment by mhilbrunner** [comment]
>But regardless, in the way I see it, if something is going to be changed (documentation/bugfix/improvement) if it does apply to 3, it does apply to 2.1 as well, with exceptions of course in those specific features/design changes that happened between the two.

AFAIK there is not much that *hasn't* changed in some way. Networking? Changed. Input? Changed. GDScript? Changed. Scenes and nodes? Changed. Physics? Changed. Rendering? Changed. Audio? Changed. Shaders? Changes. UI? Changed.

There is no single area I can think of that remained completely unchanged.

We touched most all of the docs for 3.0. Nearly all of the hundreds of commits I personally have seen wouldn't apply to 2.x without changes. Only things to come to mind is math/transform docs. And even there were some small changes.

3.0 really was a huge update and refactor. Even now, keeping the docs for 3.0 and 3.1/master up to date for changes that apply to both is real work.

It simply does not make sense for contributors to spent work on docs for a version that is going to be EOLed soon, and even more on an monumental task like this.

Of course, we'd never stop someone if that someone really wanted to do that. Still, I'm going to close this as **#wontfix** :)
